internal/formats/common/property_encoder.go

```diff
@@ -0,0 +1,325 @@
+package common
```

Comment:
Fun fact: if you take this property_encoder and it's test out of the lines changed, this is a net reduction of code.